packages/synapse-react-client/src/components/TimelinePlot/TimelinePlot.integration.test.tsx

```diff
@@ -0,0 +1,69 @@
+import { act, render, screen } from '@testing-library/react'
+import React from 'react'
+import TimelinePlot, { TimelinePlotProps } from './TimelinePlot'
+import { createWrapper } from '../../testutils/TestingLibraryUtils'
+
+import { SynapseClient } from '../../index'
+import queryResultBundleJson from '../../mocks/query/syn51735464'
+import { ColumnSingleValueFilterOperator } from '@sage-bionetworks/synapse-types'
+
+const timelineProps: TimelinePlotProps = {
+  sql: 'select * from syn123',
+  searchParams: {
+    resourceId: '9971e47e-976a-4631-8edd-5cae04304b01',
+  },
+  sqlOperator: ColumnSingleValueFilterOperator.EQUAL,
+  defaultSpecies: 'Rattus norvegicus',
+}
+
+async function renderTimeline(props: TimelinePlotProps = timelineProps) {
+  let component
+  // eslint-disable-next-line @typescript-eslint/require-await
+  await act(async () => {
+    component = render(<TimelinePlot {...props} />, {
+      wrapper: createWrapper(),
+    })
+  })
+  return component
+}
+
+describe('TimelinePlot tests', () => {
+  jest
+    .spyOn(SynapseClient, 'getFullQueryTableResults')
+    .mockResolvedValue(queryResultBundleJson)
+
+  it('renders all phases of the timeline plot', async () => {
+    await renderTimeline()
+    expect(SynapseClient.getFullQueryTableResults).toHaveBeenCalledTimes(1),
```

Comment:
I'm not convinced that `await act` reliably waits until all async side effects are done. Recommend changing this line to
`await waitFor(() => expect(SynapseClient.getFullQueryTableResults).toHaveBeenCalledTimes(1))`